corehq/apps/app_manager/views/apps.py

```diff
@@ -781,14 +780,20 @@ def rearrange(request, domain, app_id, key):
         elif "modules" == key:
             app.rearrange_modules(i, j)
     except IncompatibleFormTypeException:
-        messages.error(request, _('The form can not be moved into the desired menu.'))
+        error = _('The form is incompatible with the destination menu and was not moved.')
```

Comment:
This would be a really frustrating error message to receive... Why is it incompatible? What can I do about it?

Reply:
You can't do anything about it, unfortunately. I can make the "why" aspect clearer, though.